BHoM_Engine/Modify/SetPropertyValue.cs

```diff
@@ -192,31 +193,74 @@ public static object SetPropertyValue(this object obj, string propName, object v
         /**** Private Methods                           ****/
         /***************************************************/
 
-        private static bool SetValue(this IBHoMObject obj, string propName, object value)
+        private static bool SetValue(this IObject obj, string propName, object value)
         {
-            if (obj == null) return false;
+            if (obj == null || propName == null)
+                return false;
+            
+            // Handle dynamic objects
+            if(obj is IDynamicObject)
+            {
+                if (obj is IDynamicPropertyProvider)
+                {
+                    object result = null;
+                    bool success = Compute.TryRunExtensionMethod(obj, "SetProperty", new object[] { propName, value }, out result);
+
+                    if (success)
+                        return true;
+                }
+                else
+                {
+                    List<PropertyInfo> dynamicProperties = obj.GetType().GetProperties()
+                    .Where(x => x.GetCustomAttribute<DynamicPropertyAttribute>() != null && typeof(IDictionary).IsAssignableFrom(x.PropertyType) && x.PropertyType.GenericTypeArguments.First().IsEnum)
+                    .ToList();
 
-            if (value is IFragment)
+                    // Try to save into a dynamic property
+                    foreach (PropertyInfo prop in dynamicProperties)
+                    {
+                        if (SetValue(prop.GetValue(obj) as dynamic, propName, value))
```

Comment:
Here we might need to null check the value, and instantiate it if null. Essentially, if the dictionary is null => create a new one. (had this case happening when testing and using default constructors if not removing the input of the dictionary, as that was setting it to null).

Reply:
I would be interested to see an example where that happens as I couldn't reproduce this myself.

Reply:
Happy to leave this for now - especially as would be quite convoluted to fix with the generic types, but can be reproduced by making a faulty object where the initialisation of the dynamic dictionary on the class is not set to a new dictionary, but rather left empty or set to null. 

Could ofc argue that that is the issue of that class's implementor, rather than this method. Just if we want to sort of make this a bit safer or not. Not 100% sure tbh.

Reply:
>Could ofc argue that that is the issue of that class's implementor, rather than this method. 

Exactly!